Wip/masher

Morgiliath opened the pull request:
Improved formatting, added button toggle to turn mashing on or off.

Commit:
Initial project files.

Morgiliath committed:
Added basic arduino sketch and cad

Morgiliath committed:
Create package for masher project

Commit:
Add installation step for arduino dependencies.

Adds dependencies for building the masher project to the masher projects postinstall script.

Commit:
Add installation step for arduino dependencies.

Adds dependencies for building the masher project to the masher projects postinstall script.

Commit:
Resolve merge conflict

Morgiliath committed:
Rename masherprototype sketch to masher

Morgiliath committed:
Merge branch 'main' of https://github.com/SE-Moonbase-Operation/semoonbase into feature/masherbuild

Morgiliath committed:
Create build script to compile sketch

Morgiliath committed:
Merge pull request #1 from SE-Moonbase-Operation/feature/masherbuild

Create Build Script

Commit:
Change root project.json to npm workspace.

Changes the root of the repository to server as a npm workspace. See https://docs.npmjs.com/cli/v7/using-npm/workspaces for more information.

Commit:
Merge pull request #2 from semoonbase/feature/npm-workspace

Change root project.json to npm workspace.

Morgiliath committed:
Apply default formatting

Standardized formatting for sketch.

Morgiliath committed:
Implement Button Toggle

Changes from the initial implementation of mashing only when the button is held down, to having an internal state activated by pressing the button. Also cleaned up some comments and moved some actions to occur when the internal state is toggled.

Morgiliath committed:
Added delay and a release for keypress

new variable: mashDelay , sets how long the key is held and released for. Used that to make sure we aren't just holding keys down indefinitely.

Morgiliath committed:
Fixed Formatting

Morgiliath committed:
Fix Build script and add flash script

build script had the wrong file path and would give an error, added flash script (may have issues because the serial port varies)

Commit:
Make windows and linux flash scripts.

Changes the flash script to be platform dependant.

Commit:
Move package lock to workspace dir.

Moves the node package lock into the root of the workspace, where it is conventionally located.

Commit:
Update .gitigore.

Add's Node's module cache to .gitignore.

Commit:
Refactor GPIO Assignment code.

Refactors GPIO assignments. Changes global constants to global enum. Removes some code repitition.

Commit:
Refactor button toggling code.

Refactors button toggling code into its own function.

Commit:
Refactor key selection.

Refactors key selection code into own function.

Commit:
Refactor mashing logic.

Refactors mashing code into high level API which allows consumer to ignore key delays.

Commit:
Create enum to store keyboard keys.

Commit:
Create npm dev script for masher.

Creates a dev scirpt for masher that automatically rebuilds, reflashes, and tests the key output of the device.

Commit:
Update firmware assignents of GPIO inputs.

Commit:
Seperate masher source code from rest of project.

Commit:
Ignore IDE Configs.

Ignores the project configuration folders for common IDEs.

Commit:
Modularize GPIO code.

Commit:
Add script for rapid iteration of masher code.

Commit:
Modularize projects by concern.

Commit:
Move pin assignment into GPIO module.

Commit:
refactor to use an undefined module interface.

Commit:
Minor refactoring.

Commit:
Scaffold support for reading GPIO into vendor target.

Commit:
Update .gitignore for manual build.

freylint commented on projects/masher/package.json:
```diff
@@ -8,7 +8,11 @@
     "install-toolchain": "arduino-cli core install arduino:avr",
     "install-lib": "arduino-cli lib install Keyboard",
-    "build": "arduino-cli compile -b arduino:avr:leonardo"
+    "build": "arduino-cli compile -b arduino:avr:leonardo firmware",
```
Add missing HID dependency to arduino-cli dep install.

Commit:
Add missing HID library install.

Fixes issues with distributions of arduino that do not include the HID library.

Commit:
Update .gitignore for multiboard build.

Commit:
Shadow Makefile Library for building Arduino sketches.

Commit:
Create makefile for masher build using native toolchain.

Commit:
Update project for native toolchain build.

Commit:
Create npm script for building using native toolchain.

Commit:
Change iter script to use native toolchain.

Will allow test integration at a later date.

Commit:
Add unit testing module to libs.

Commit:
Remove preinstalled HID dependency.

Fixes issue blockingg npm install.

Commit:
Add catch3 build script.

Adds a script to build catch3 from source using bazelisk.

Commit:
Shadow full arduino-makfile project.

Commit:
Add submodule init to npm install script.

Commit:
Remove manually shadows arduino-makefile.

Commit:
Switch semasher to auto shadowed arduino-makeifle.

Commit:
Update .gitignore for bazel and pio.

Commit:
Restructure masher for bazel.

Commit:
Scaffold library for key mocking.

Morgiliath committed:
Merge pull request #4 from semoonbase/wip/bazel-experiment

Change build system to bazel.

Commit:
Fix mockkey build.

Fixes mockkey build by correcting paths to the sources that were intended to be built.

Commit:
Write naive implementation of the Arduino Keyboard library.

Commit:
Restructure masher library into own top level project.

Commit:
Format project bazel files.

Commit:
Update platformio configurations.

Commit:
Add coverage context for base LICENSE.md.

Commit:
Apply new libmasher structure to the build system.

Commit:
Update masher firmware build to use new libmasher structure.

Commit:
Restructure libmasher to remove src directory.

Commit:
Remove redundant projects directory.

Commit:
Fix mockkey header export.

Fixes issue in mockkey build definition that prevented the headers from being included in dependents.

Commit:
Restructure libmasher for platformio compat.

Commit:
Split mockard components into seperate libraries.

Commit:
Add native lib version of masher libs.

Morgiliath committed:
Merge pull request #6 from semoonbase/wip/unit-testing

Merge basic native build of masher libraries into wip/masher

Commit:
Add main to masher core.

Adds a main function to the masher core library. This is solely intended to serve as an entry point for gtest.

Commit:
Set up google test for masher project.

Commit:
Correct doc comment in masher's core.cc.

Commit:
Happy path tests for core mapKey function.

Commit:
Create build def for core's tests.

Commit:
Set up tests for key module.

Commit:
Fix functionality of the kbemu module setup interface test.

Commit:
Add unit testing to gpio module.

Commit:
Add unit testing support to core module.

Commit:
Remove PIO test directory.

Commit:
Remove redundant include gaurds.

Commit:
Add bazel target for core module.

Commit:
Create README.md

Commit:
Format README.md

Morgiliath committed:
Merge pull request #7 from semoonbase/wip/unit-testing

Add unit testing to masher dev branch.

Commit:
Scaffold development container.

Commit:
Merge branch 'main' of github.com:semoonbase/semoonbase into wip/masher

Commit:
Update devonctainers to use base debian install.

Commit:
Add missing deps to devcontainer.

Commit:
Add ibazel to dev container.

Commit:
Switch to full debian for dev container.

Commit:
Add explicit openssh-clientt dependency to devcontainer.

Commit:
Fix inclusion of mapKey tests.

Commit:
Scaffold gpio to key mapping function.

Commit:
Implement basic switching functionality.

freylint committed:
Switch core gpio to key map func to Arduino HIGH.
